src/Domain/DonationStatus.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace MatchBot\Domain;
+
+/**
+ * @link https://docs.google.com/document/d/11ukX2jOxConiVT3BhzbUKzLfSybG8eie7MX0b0kG89U/edit?usp=sharing
+ * @todo Consider vs. Stripe options
+ *
+ * @todo before merging BG2-2296 - check that the prod DB doesn't have any statuses recorded not listed below.
+ */
+enum DonationStatus: string
+{
+    public const SUCCESS_STATUSES = [self::Collected, self::Paid];
+
+    public const NEW_STATUSES = [self::NotSet, self::Pending];
+
+    /**
+     * @link https://thebiggive.slack.com/archives/GGQRV08BZ/p1576070168066200?thread_ts=1575655432.161800&cid=GGQRV08BZ
+     */
+    public const REVERSED_STATUSES = [self::Refunded, self::Failed, self::Chargedback];
+
```

Comment:
Comparing the below list of statues to the list of possible statuses that were on the Donation class (copied here for ref):

```

        'Pending',
        'Collected',
        'Paid',
        'Cancelled',
        'Refunded',
        'Failed',
        'Chargedback',
        'RefundingPending',
        'PendingCancellation',
```

The following ones were not in the Donation class list: *NotSet*, *Reserved* but they are below because they are mentioned in code so they have to be here. We can delete them if/when we delete the code that references them. SA checks don't allow the references to exist without the cases existing below.

Reply:
Did an SQL query to check all the donation statuses we have in the database - anything that doesn't exist in the database and we don't have code to create in future could be deleted from the Enum. DB results:

`SELECT DISTINCT donationStatus FROM `Donation`; `
```
"Cancelled"
"Pending"
"Collected"
"Refunded"
"Paid"
"Failed"
"Chargedback"
```